### ISSUE TITLE
Incorrectly reporting `No overloads for "round" match the provided arguments`

### ISSUE THREAD

**Issue by ericroberts** [microsoft/pyright#9153]
**Describe the bug**

When wrapping calls to `round` that contain two arguments with `int` subsequent calls to round will error saying `No overloads for "round" match the provided arguments`.

**Code or Screenshots**

```python
int(round(1.2, 0))
round(3.4, 2)
```

Pyright errors on the second line here with version 1.1.383. If I remove the second argument to round, or I remove the wrapping `int()` the problem goes away.

For example this is fine:

``` python
round(1.2, 2)
round(3.4, 2)
```

and this is fine:

``` python
int(round(1.2))
round(3.4, 2)
```

In our codebase the two uses of round were in different files, took me a while to figure out and produce this example.

**VS Code extension or command-line**

Command line, version 1.1.383

**Comment by erictraut** [comment]
Thanks for isolating the problem to a concise repro case.

I suspect this is related to #9132 and #9130. I'll investigate further.
